### DISCUSSION TITLE
Install coqui-ai TTS in Conda on Windows tutorial

### DISCUSSION THREAD

**Discussion by tacitus-git** [coqui-ai/TTS#2373] [Show and tell]
clone git
conda create -n coquitts python=3.10
conda activate coquitts
conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
cd (directory of tts)
pip install -r requirements.txt
python setup.py develop
#use python script to produce tts results

This is not a detailed tutorial, but it is damn better than what I had. Hopefully this will help someone.

**Comment by king-dahmanus** [comment]
What about a tutorial for those without a GPU?

**Reply by casiimir** [reply]
You can use the CPU flag, setting torch too:

```python
import torch
from TTS.api import TTS

device = torch.device("cpu")
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", gpu=False)
```

**Comment by Hexatona** [comment]
Awesome - I had to wade through an hour long video just to figure all this out.  You're very to the point - appreciated.

**Comment by Cem-Kaya** [comment]
thank you, this also worked for me 

git clone https://github.com/coqui-ai/TTS # clone repo 
conda create -n coquitts python=3.10 # create python env 
conda activate coquitts
conda install pytorch torchvision torchaudio pytorch-cuda=12.1 -c pytorch -c nvidia # works with cuda 12.1 
cd (directory of tts)
pip install -e .[all,dev,notebooks]  # adding  notebooks requirements fixed a bug in notebooks 
python setup.py develop

**Reply by bobsteele** [reply]
As far as I know, only Nvidia offers CUDU in their GPU's.  If that is the case, then its suggesting that TTS requires a Nvidia GPU. Am I correct in my understanding?

**Reply by TheMechanicX32** [reply]
No. There are instructions on the pytorch website for installing pytorch on CPU-only devices. I am running xtts on a 6-year-old MacBook Pro perfectly fine. 

Please hit me up if anyone has any questions.

**Comment by jj449** [comment]
> clone git conda create -n coquitts python=3.10 conda activate coquitts conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia cd (directory of tts) pip install -r requirements.txt python setup.py develop #use python script to produce tts results
> 
> This is not a detailed tutorial, but it is damn better than what I had. Hopefully this will help someone.

Thank you  so much , very useful . Installed successfully according this !

**Comment by Panquesito0** [comment]
I tried your method using a MacBook Air M1. After creating an enviornment on Conda using Python 3.9.19 with the name "ai_stuff". 
I continue with the next steps:

git clone https://github.com/coqui-ai/TTS 
conda activate ai_stuff
cd (directory of tts)
pip install -r requirements.txt
python setup.py develop

Thanks. Really helpful.

**Comment by VictorJulianiR** [comment]
for those that don't want to install conda, this line did the trick for me:

`pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121`

**Reply by TheMechanicX32** [reply]
I totally understand the desire to avoid Conda, but conda is really useful for building environments for XTTS or something along those lines. Allows me to create an environment where I can install very specific versions of packages without affecting my other environments.
